ee/tabby-webserver/src/service/email.rs

```diff
@@ -24,4 +25,17 @@ impl EmailService for DbConn {
     async fn delete_email_setting(&self) -> Result<()> {
         self.delete_email_setting().await
     }
+
+    async fn send_mail(&self, messages: &mut Vec<Message>) -> Result<()> {
```

Comment:
```suggestion
    async fn send_mail(&self, messages: Message) -> Result<()> {
```

Reply:
I made it a `&mut Vec<Message>` because it has to retrieve the SMTP credentials from the database every time, and establish a new connection to the server (with DNS). I wanted to allow batching, and also have a way to know which messages weren't sent if an error occurs. Do we not plan to ever send multiple emails at once? (i.e. announcement emails sent to all users)

Reply:
> it has to retrieve the SMTP credentials from the database every time, and establish a new connection to the server (with DNS).
Good point - in that case, I'll really prefer MailService being something individually initialized (thus maintains its connection by itself), and recreate connection whenever the credentials being updated in database (certain debouncing required)

> Do we not plan to ever send multiple emails at once? (i.e. announcement emails sent to all users)
Unlikely. The email service is mostly used for sending invitation / lost password / ...